seametrics/payload/processor.py

```diff
@@ -85,18 +85,71 @@ def compute_payload(self) -> Payload:
 
     def validate_input_parameters(self, dataset_name: str):
         """
-        Validates the input parameters.
+    Validates the input parameters.
+
+    Args:
+        dataset_name (str): The name of the dataset.
+
+    Raises:
+        TypeError: If any of the parameters are of the wrong type.
+        ValueError: If dataset is not found in FiftyOne or if tracking mode is enabled for RGB data.
+
+    Validations:
+        - `dataset_name` must be a string.
+        - `gt_field` must be a string.
+        - `models` must be a list of strings.
+        - `tracking_mode` must be a boolean.
+        - `sequence_list` must be None or a list of strings.
+        - `data_type` must be either "rgb" or "thermal".
+        - `excluded_classes` must be None or a list of strings.
+        - `slices` must be None or a list of strings.
+        - `tags` must be None or a list of strings
+        """
+        
+        # Check dataset_name is a string
+        if not isinstance(dataset_name, str):
+            raise TypeError(f"dataset_name must be of type str, but got {type(dataset_name)}")
 
-        Args:
-            dataset_name (str): The name of the dataset.
+        # Check gt_field is a string
+        if not isinstance(self.gt_field, str):
+            raise TypeError(f"gt_field must be of type str, but got {type(self.gt_field)}")
 
-        Raises:
-            ValueError: If dataset is not found or tracking mode enabled for RGB data.
-        """
+        # Check models is a list of strings
+        if not isinstance(self.models, list) or not all(isinstance(model, str) for model in self.models):
+            raise TypeError(f"models must be a list of strings, but got {self.models}")
+
+        # Check tracking_mode is a boolean
+        if not isinstance(self.tracking_mode, bool):
+            raise TypeError(f"tracking_mode must be of type bool, but got {type(self.tracking_mode)}")
+
+        # Check sequence_list is None or a list of strings
+        if self.sequence_list is not None and (not isinstance(self.sequence_list, list) or not all(isinstance(seq, str) for seq in self.sequence_list)):
+            raise TypeError(f"sequence_list must be a list of strings or None, but got {self.sequence_list}")
+
+        # Check data_type is either "rgb" or "thermal"
+        if self.data_type not in ["rgb", "thermal"]:
+            raise ValueError(f"data_type must be 'rgb' or 'thermal', but got {self.data_type}")
+
+        # Check excluded_classes is None or a list of strings
+        if self.excluded_classes is not None and (not isinstance(self.excluded_classes, list) or not all(isinstance(cls, str) for cls in self.excluded_classes)):
+            raise TypeError(f"excluded_classes must be a list of strings or None, but got {self.excluded_classes}")
```

Comment:
This is being very picky, but if we are throwing `TypeError`s it would be good to print the `{type(self.xxx)}` instead of the value itself like you do in previous `TypeError` checks.